src/System/IO/Resource/Linear/Internal.hs

```diff
@@ -129,50 +140,34 @@ hClose :: Handle %1 -> RIO ()
 hClose (Handle h) = unsafeRelease h
 
 hIsEOF :: Handle %1 -> RIO (Ur Bool, Handle)
-hIsEOF = coerce (unsafeFromSystemIOResource System.hIsEOF)
+hIsEOF = unsafeFromSystemIOHandle System.hIsEOF
 
 hGetChar :: Handle %1 -> RIO (Ur Char, Handle)
-hGetChar = coerce (unsafeFromSystemIOResource System.hGetChar)
+hGetChar = unsafeFromSystemIOHandle System.hGetChar
 
 hPutChar :: Handle %1 -> Char -> RIO Handle
-hPutChar h c = flipHPutChar c h -- needs a multiplicity polymorphic flip
-  where
-    flipHPutChar :: Char -> Handle %1 -> RIO Handle
-    flipHPutChar c =
-      coerce (unsafeFromSystemIOResource_ (\h' -> System.hPutChar h' c))
+hPutChar = flip (\c -> unsafeFromSystemIOHandle_ (\h' -> System.hPutChar h' c))
 
 hGetLine :: Handle %1 -> RIO (Ur Text, Handle)
-hGetLine = coerce (unsafeFromSystemIOResource Text.hGetLine)
+hGetLine = unsafeFromSystemIOHandle Text.hGetLine
 
 hPutStr :: Handle %1 -> Text -> RIO Handle
-hPutStr h s = flipHPutStr s h -- needs a multiplicity polymorphic flip
-  where
-    flipHPutStr :: Text -> Handle %1 -> RIO Handle
-    flipHPutStr s =
-      coerce (unsafeFromSystemIOResource_ (\h' -> Text.hPutStr h' s))
+hPutStr = flip (\s -> unsafeFromSystemIOHandle_ (\h' -> Text.hPutStr h' s))
 
 hPutStrLn :: Handle %1 -> Text -> RIO Handle
-hPutStrLn h s = flipHPutStrLn s h -- needs a multiplicity polymorphic flip
-  where
-    flipHPutStrLn :: Text -> Handle %1 -> RIO Handle
-    flipHPutStrLn s =
-      coerce (unsafeFromSystemIOResource_ (\h' -> Text.hPutStrLn h' s))
+hPutStrLn = flip (\s -> unsafeFromSystemIOHandle_ (\h' -> Text.hPutStrLn h' s))
 
 -- | See @System.IO.'System.IO.hSeek'@.
 --
 -- @since 0.3.0
 hSeek :: Handle %1 -> System.SeekMode -> Integer -> RIO Handle
-hSeek h mode i = coerce hSeek' h
-  where
-    hSeek' :: Handle %1 -> RIO Handle
-    hSeek' =
-      coerce (unsafeFromSystemIOResource_ (\h' -> System.hSeek h' mode i))
+hSeek h mode i = unsafeFromSystemIOHandle_ (\h' -> System.hSeek h' mode i) h
```

Comment:
why don't we need a `flip` here? Or why do we need it in the other functions?

Reply:
There is more than two arguments, so `flip` doesn't work (well, doesn't suffice, really).

Maybe it would have been more uniform to define all the functions in this style, instead of using `flip`. I just read the comment that we'd left, and just used flip without really thinking.

What looks better?
```haskell
 hPutStrLn = flip (\s -> unsafeFromSystemIOHandle_ (\h' -> Text.hPutStrLn h' s))
 hPutStrLn h s = unsafeFromSystemIOHandle_ (\h' -> Text.hPutStrLn h' s) h
```

Reply:
The second one looks better for me. At first I tought that `flip` did something smart here, but then I saw this last function that seemed clearer to me.

Reply:
I removed flip.